### ISSUE TITLE
Dependency loop in latest tag

### ISSUE THREAD

**Issue by niallnsec** [BurntSushi/toml#312]
Hello,

It appears that the new v0.4.0 tag has introduced a fatal dependency loop which is prevent any modules which reference this repo from updating. The output from go get -u is as follows:

```
go get: github.com/BurntSushi/toml@v0.3.1 updating to
        github.com/BurntSushi/toml@v0.4.0 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210723065233-facb9eccd4da requires
        github.com/BurntSushi/toml@v0.3.2-0.20210704081116-ccff24ee4463 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210704062846-269931e74e3f requires
        github.com/BurntSushi/toml@v0.3.2-0.20210624061728-01bfc69d1057 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210624055653-1f6389604dc6 requires
        github.com/BurntSushi/toml@v0.3.2-0.20210621044154-20a94d639b8e
```

**Comment by niallnsec** [comment]
I should also add the problem does persist in the current master as well, even after the updated version of toml-test:
```
go: github.com/BurntSushi/toml@v0.4.1-0.20210804092126-444cd1471c30 requires
        github.com/BurntSushi/toml-test@v1.0.0 requires
        github.com/BurntSushi/toml@v0.4.0 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210723065233-facb9eccd4da requires
        github.com/BurntSushi/toml@v0.3.2-0.20210704081116-ccff24ee4463 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210704062846-269931e74e3f requires
        github.com/BurntSushi/toml@v0.3.2-0.20210624061728-01bfc69d1057 requires
        github.com/BurntSushi/toml-test@v0.1.1-0.20210624055653-1f6389604dc6 requires
        github.com/BurntSushi/toml@v0.3.2-0.20210621044154-20a94d639b8e
```

**Comment by arp242** [comment]
Yeah, this is being discussed at https://github.com/BurntSushi/toml-test/issues/74 as well.

I'll look this afternoon; it does work most of the time, but not in some cases.

**Comment by vissong** [comment]
go: github.com/BurntSushi/toml@v0.4.0 requires
	github.com/BurntSushi/toml-test@v0.1.1-0.20210723065233-facb9eccd4da requires
	github.com/BurntSushi/toml@v0.3.2-0.20210704081116-ccff24ee4463 requires
	github.com/BurntSushi/toml-test@v0.1.1-0.20210704062846-269931e74e3f requires
	github.com/BurntSushi/toml@v0.3.2-0.20210624061728-01bfc69d1057 requires
	github.com/BurntSushi/toml-test@v0.1.1-0.20210624055653-1f6389604dc6 requires
	github.com/BurntSushi/toml@v0.3.2-0.20210621044154-20a94d639b8e: invalid version: unknown revision 20a94d639b8e

same problem

**Comment by arp242** [comment]
Fixed in #313, and tagged v0.4.1 with this fix.
